### PR TITLE
Extends ShuffleAndSplitJob: allows gzip files as input and output

### PR DESCRIPTION
**Pull request by christophmluscher**
Extends the functionality of the `ShuffleAndSlpitJob` to process gzipped file input and output a gzipped file allowing for a wider use of the job.

### CODE REVIEW

**Review by NeoLegends — APPROVED** [review]

**Comment by NeoLegends** [comment]
Nit: I think we often call the init flag `compressed` rather than `zip_output`.

File: corpus/segments.py [hunk 1]
```diff
@@ -172,13 +175,16 @@ def __init__(self, segment_file, split=None, shuffle=True, shuffle_seed=0x3C5EA3
         self.shuffle = shuffle
         self.shuffle_seed = shuffle_seed
 
-        self.out_segments = {k: self.output_path("%s.segments" % k) for k in self.split.keys()}
+        self.out_segments = {
+            k: self.output_path("%s.segments" % k if not zip_output else "%s.segments.gz" % k)
```
Comment (michelwi) [inline corpus/segments.py:179]:
```suggestion
            k: self.output_path(f"{k}.segments{'.gz' if zip_output else ''}")
```
shorter + use f-string ?

**Review by michelwi — APPROVED** [review]
just one optional comment for style.. otherwise lgtm

**Comment by christophmluscher** [comment]
> Nit: I think we often call the init flag `compressed` rather than `zip_output`.

sadly I found both..

**Pull request merged**